### ISSUE TITLE
ES6 is not transpiled in addon

### ISSUE THREAD

**Issue by aceofspades** [ember-cli/ember-cli#3556]
Per https://github.com/babel/ember-cli-babel/issues/13#issuecomment-81176286 it was stated that ES6 in addons should be transpiled the same as with the hosting app.

This example has a simple component definition in both the app and addon, with an ES6 fat arrow function. For the app, it is transpiled, but not in the addon.

To see the issue:

```
git clone git@github.com:aceofspades/es6-in-addon.git
cd es6-in-addon/alpha
ember install
ember build
```

In `alpha/dist/assets/alpha.js` the function is transpiled:
```
double: function (v) {
  return v * 2;
}
```

In `alpha/dist/assets/vendor.js` it is not.
```
double: (v => v * 2)
```

**Comment by stefanpenner** [comment]
Addons the addon needs ember-cli-babel

**Comment by rwjblue** [comment]
https://github.com/aceofspades/es6-in-addon/blob/master/alpha/package.json#L25  needs to be in deps, not dev deps.

**Comment by jayphelps** [comment]
@rwjblue That's the default place it's put when `$ ember addon foo` since it shares the package.json from app blueprint. This just bit me as well, lost 2+ hours trying to figure out why addon tests were failing/hanging was because they weren't getting transpiled. This isn't obvious--should we have this added for addons automatically? Seems like a bug that ember apps have babel by default but addons don't.

**Comment by rwjblue** [comment]
@jayphelps -  This is fundamentally how npm works, if you have a devDep it is installed when working on the project/lib but not when it is used as a dependency of something else. 

I am not opposed to shipping babel in addons by default just like we do with apps.

**Comment by stefanpenner** [comment]
> I am not opposed to shipping babel in addons by default just like we do with apps.

it feels like we actually have 2 types of addons,

1. small addon, that provides 1 simple thing (maybe a command for ember-cli, or jshinting)
2. an addon that provides addon/ and app/

these seem to have different dep requirements, it would be strange for 1 to have babel as a dep, but makes total sense for 2

thoughts?

**Comment by jschilli** [comment]
+1 wonder if having sub blueprints for build-addon and ?? functional-addon (terrible name) would make sense

**Comment by stefanpenner** [comment]
the UX feels crappy around this, needs some exploration.

**Comment by aceofspades** [comment]
Thanks for the help.

I would vote for consistent UX app vs. addon, maybe an option `--lightweight` when creating an addon for those that are concerned about this. 

I realize there may be some slight reformatting, but are there other significant undesirable side effects to having babel, for example, by default in an addon when it is not actually used? Isn't the focus on ES6/2015 going forward?

(Maybe the `/* jshint esnext: true */` flag should be used to trigger ES6 transpilation.)

**Comment by jayphelps** [comment]
For anyone watching this thread, this is was fixed in #3564
